platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -294,13 +294,56 @@ + (instancetype)mgl_predicateWithJSONObject:(id)object {
+                NSString *leftOperator = leftCondition.firstObject;
+                NSString *rightOperator = rightCondition.firstObject;
+                
+                NSArray *limits;
+                NSExpression *leftConditionExpression;
+                
+                if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@"<="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).rightExpression, ((NSComparisonPredicate *)subpredicates[1]).rightExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).leftExpression;
+                
+                } else if ([leftOperator isEqualToString:@"<="] && [rightOperator isEqualToString:@"<="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).leftExpression, ((NSComparisonPredicate *)subpredicates[1]).rightExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).rightExpression;
+                
+                } else if([leftOperator isEqualToString:@"<="] && [rightOperator isEqualToString:@">="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).leftExpression, ((NSComparisonPredicate *)subpredicates[1]).leftExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).rightExpression;
+                
+                } else if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@">="]) {
+                    limits = @[((NSComparisonPredicate *)subpredicates[0]).rightExpression, ((NSComparisonPredicate *)subpredicates[1]).leftExpression];
+                    leftConditionExpression = ((NSComparisonPredicate *)subpredicates[0]).leftExpression;
+                }
+                
+                if (limits && leftConditionExpression) {
+                     return [NSPredicate predicateWithFormat:@"%@ BETWEEN %@", leftConditionExpression, [NSExpression expressionForAggregate:limits]];
+                }
+            }
+        }
         return [NSCompoundPredicate andPredicateWithSubpredicates:subpredicates];
     }
     if ([op isEqualToString:@"any"]) {
         NSArray *subpredicates = MGLSubpredicatesWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
         return [NSCompoundPredicate orPredicateWithSubpredicates:subpredicates];
     }
+    if ([op isEqualToString:@"match"]) {
+        NSExpression *expression = [NSExpression expressionWithMGLJSONObject:object];
+        return [NSComparisonPredicate predicateWithLeftExpression:expression
+                                                  rightExpression:[NSExpression expressionForConstantValue:@YES]
+                                                         modifier:NSDirectPredicateModifier
+                                                             type:NSNotEqualToPredicateOperatorType
```

Comment:
I think we want `["match", ["get", "a"], 1, true, 2, true, 3, true, 4, true, false]` to end up as `MGL_MATCH(a, 1, true, 2, true, 3, true, false) == true`, so the type should be `NSEqualToPredicateOperatorType`.